language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SnippetGenerator.java

```diff
@@ -35,7 +35,7 @@
  */
 public class SnippetGenerator {
 
-    private static final String FILTER_TEXT_SEPARATOR = "|";
+    private static final String FILTER_TEXT_SEPARATOR = "_";
```

Comment:
Just wanted to know why have we removed "|".

Reply:
That is because, if we have non-word symbols, then the completion's filter text will honour only the first component. for example, consider,
`service|http` as the filter text. Then, even when we are trying to type `http`, the http service snippet should get filtered. When we are having the pipe in between, for the `http` part it won't show the completion.

In order to make this happen, we have to separate these components with a word character. so went with the `_`